src/main/java/com/royalrangers/service/UserService.java

```diff
@@ -268,6 +270,17 @@ public void updateUserById(Long id, UserUpdateDto update) {
         userRepository.save(user);
     }
 
+    public ResponseResult getUsersByPlatoon(){
+        String email = getAuthenticatedUserEmail();
+        User user = userRepository.findByEmail(email);
+        List<User> users = userRepository.findUsersByApprovedTrueAndPlatoon_Id(user.getPlatoon().getId());
+        List<UserProfileDto> result = new ArrayList<>();
+        for(User usr : users ){
+            result.add(buildUserProfile(usr));
+        }
+        return ResponseBuilder.success(result);
```

Comment:
I think it's wrong to return an ResponseResult here, it must build in controller.